numpy/core/tests/test_multiarray.py

```diff
@@ -2285,7 +2285,9 @@ def test_searchsorted_unicode(self):
     def test_searchsorted_with_sorter(self):
         a = np.array([5, 2, 1, 3, 4])
         s = np.argsort(a)
-        assert_raises(TypeError, np.searchsorted, a, 0, sorter=(1, (2, 3)))
+        with assert_raises(TypeError):
+            with assert_warns(DeprecationWarning):
+                np.searchsorted(a, 0, sorter=(1, (2, 3)))
```

Comment:
Does this actually work? Is there some easy proof?

Reply:
LGTM. It first warns, then raises a typeerror due to mismatch. Adding the `assert_warns` means that the context manager makes sure the DeprecationWarning is not caught.

Reply:
Sorry, the inner with statement never gets around to checking that the warning was raised. I wonder if that it in a sense a small issue with it. I am OK as is. It still tests the original usecase, a `sorter` failing to convert to an array entirely is hopefully covered elsewhere.

Could be split into two tests, one with setting to error explicitely.

Reply:
yes, 772f689